.golangci.yml

```diff
@@ -24,6 +24,17 @@ linters-settings:
     require-explanation: true
     require-specific: true
 
+  cyclop:
+    max-complexity: 15
+
+issues:
+  exclude-rules:
+    - path: (.+)_test.go
+      linters:
+        - funlen
+        - dupl
+        - cyclop
```

Comment:
what do you think about length and duplication checks?

Reply:
Both of those reasonable functions had a complexity of 12-15.  Maybe we bump the limit to 25?

Reply:
I also think funlen/dupl exclusion makes sense for tests at this point

Reply:
Funlen is set to 100 instructions currently, while the [TestComposeMockValueBySchema has 477](https://github.com/opentofu/opentofu/actions/runs/9100211577/job/25014502286#step:9:156) (because of the list of test-cases).

I can (1) introduce some generator functions or (2) split it into separate tests. I don't like either option, because the current one is the most readable IMO (just literal values of inputs and expected outputs). Some of these inputs / outputs are duplicated (but are different test-cases) thus triggering dupl issues.

The test function is really long and I wouldn't pass something that lengthy as a non-test code so I would still prefer to skip these linters (funlen, dupl) for tests and keep their existing limits for non-testing code.

Regarding the complexity limit - 25 sounds ok.

Reply:
I think for tests that's just busywork?  Maybe we disable for now and revisit this once we have a better idea of how we want the test structure to work?